google-cloud-storage/src/test/java/com/google/cloud/storage/it/GrpcPlainRequestLoggingInterceptor.java

```diff
@@ -59,6 +75,19 @@ public final class GrpcPlainRequestLoggingInterceptor implements ClientIntercept
   private static final Metadata.Key<String> X_GOOG_REQUEST_PARAMS =
       Metadata.Key.of("x-goog-request-params", Metadata.ASCII_STRING_MARSHALLER);
 
+  private static final Map<String, Message> anyParsers =
```

Comment:
This appears clever, which makes me suspicious, but it also seems like the most concise way to do this.

Reply:
I'll add a comment to clarify what it is doing. But the code is actually straight forward (given N message types, map their `type_url`s to the types) but has some java boilerplate to it because of protobuf-java's weird static/non-static API.